pkg/providers/tinkerbell/upgrade.go

```diff
@@ -292,6 +308,30 @@ func (p *Provider) ValidateNewSpec(ctx context.Context, cluster *types.Cluster,
 	return nil
 }
 
+func validateWorkerNodeGroupMachineConfigRefsUnchanged(current, desired []v1alpha1.WorkerNodeGroupConfiguration) error {
```

Comment:
Not PR blocking; I want to voice my opinion again on the naming of this function. A shorter name like: `validateRefsUnchanged` provides just as much information/value to understanding this function as the current name but it's significantly shorter. This is valuable for a few reasons. It improves the readability, avoids repetition of input parameter values, it doesn't perpetuate the pattern of very long naming, it more closely adheres to the stdlib naming patterns, and promotes code comments for greater understanding, etc.

While not normative or canonical, this style guide from Google also describes this idea well: https://google.github.io/styleguide/go/best-practices#function-and-method-names

Reply:
This is probably fine to do now. With a previous iteration of the PR there was a collision between `validateControlPlaneMachineRefsUnchanged` and `validateWorkerNodeGroupMachineRefsUnchanged` (the Google styleguide points this out as an exception). 

Naming in bloated packages is always going to be difficult because there is so much in-scope. While I agree, I get dubious real fast in packages like `tinkerbell`.

Reply:
Updated